src/migratePivotFolder.ts

```diff
@@ -0,0 +1,179 @@
+import { ContentRecord } from "@activeviam/activeui-sdk";
+import xml2js from "xml2js";
+import _cloneDeep from "lodash/cloneDeep";
+import { generateId } from "./generateId";
+
+/**
+ * Extracts and parses the XML calculated measure objects from the given pivot folder.
+ */
+const getCalculatedMeasuresFromPivotFolder = async (
+  pivotFolder: ContentRecord
+): Promise<
+  {
+    expression: string;
+    formatStringExpression: string;
+    uniqueName: string;
+    id: string;
+  }[]
+> => {
+  const parser = new xml2js.Parser();
+
+  const calculatedMeasures: {
+    expression: string;
+    formatStringExpression: string;
+    uniqueName: string;
+    id: string;
+  }[] = [];
+
+  if (pivotFolder?.children) {
+    await Promise.all(
+      Object.values(pivotFolder?.children).map(async ({ children = {} }) => {
+        const calculatedMeasureEntries = Object.entries(children);
+
+        const measures = await Promise.all(
+          calculatedMeasureEntries.map(async (cmEntry) => {
+            const [_, entry] = cmEntry;
+            const result = await parser.parseStringPromise(entry.entry.content);
+            const id = generateId();
```

Comment:
For separation of concerns, I wouldn't generate the id in this function: keep the role of this function of just parsing the necessary info, and then use that info together with a generated id only outside the function.